src/pepr/operator/reconcilers/package-reconciler.ts

```diff
@@ -66,62 +70,100 @@ export async function packageReconciler(pkg: UDSPackage) {
     await new Promise(resolve => setTimeout(resolve, backOffSeconds * 1000));
   }
 
-  // Migrate the package to the latest version
-  migrate(pkg);
-
-  // Configure the namespace and namespace-wide network policies
   try {
     await updateStatus(pkg, { phase: Phase.Pending, conditions: getReadinessConditions(false) });
+    await reconcilePackageFlow(pkg);
+  } catch (err) {
+    await handleFailure(err, pkg);
+  }
+}
+
+/**
+ * Orchestrates the main reconciliation flow for a package.
+ * Handles status updates, resource creation, and sequencing.
+ */
+async function reconcilePackageFlow(pkg: UDSPackage): Promise<void> {
+  const metadata = pkg.metadata!;
+  const generation = (pkg.metadata?.generation ?? 0).toString();
+  const { namespace } = metadata;
 
-    // Get the requested service mesh mode, default to sidecar if not specified
-    const istioMode = pkg.spec?.network?.serviceMesh?.mode || Mode.Sidecar;
+  // Get the requested service mesh mode, default to sidecar if not specified
+  const istioMode = pkg.spec?.network?.serviceMesh?.mode || Mode.Sidecar;
 
-    // Pass the effective Istio mode to the networkPolicies function
-    const netPol = await networkPolicies(pkg, namespace!, istioMode);
+  // 1. First, ensure network policies are in place
+  const netPol = await networkPolicies(pkg, namespace!, istioMode);
+  const authPol = await generateAuthorizationPolicies(pkg, namespace!, istioMode);
 
-    const authPol = await generateAuthorizationPolicies(pkg, namespace!, istioMode);
+  // 2. Now enable Istio injection (this may restart pods in sidecar mode)
+  await enableIstio(pkg);
 
-    let endpoints: string[] = [];
-    // Update the namespace to enable the expected Istio mode (sidecar or ambient)
-    await enableIstio(pkg);
+  // 3. If in ambient mode, set up waypoints after network policies are in place
+  if (istioMode === Mode.Ambient) {
+    const authServiceClients = pkg.spec?.sso?.filter(sso => sso.enableAuthserviceSelector) || [];
 
-    let ssoClients = new Map<string, Client>();
-    let authserviceClients: string[] = [];
+    for (const client of authServiceClients) {
+      try {
+        log.info(`Setting up ambient waypoint for client`, {
+          namespace: pkg.metadata?.namespace,
+          package: pkg.metadata?.name,
+          clientId: client.clientId,
+          selector: JSON.stringify(client.enableAuthserviceSelector),
+        });
+        await setupAmbientWaypoint(pkg, client.clientId);
```

Comment:
When we setup the waypoint we should also ideally update any existing pods/services to use it. Right now we have the mutation that handles any pods/services coming _after_ the waypoint. But if the app is already deployed, the waypoint wouldn't be used until the pods next update/change (and go through the mutation). We probably need a function that attempts to get any matching pods/services and updates them to handle this timing?